File: Iniciante/1094.py
total = 0
total_coelhos = 0
total_ratos = 0
total_sapos = 0
num = int(input())
for i in range(num):
    n1, n2 = input().split()
    n1 = int(n1)
    n2 = str(n2).upper()
    total += n1
    if n2 == 'C':
        total_coelhos += n1
    elif n2 == 'R':
        total_ratos += n1
    elif n2 == 'S':
        total_sapos += n1
print('Total: {} cobaias'.format(total))
print('Total de coelhos: {}'.format(total_coelhos))
print('Total de ratos: {}'.format(total_ratos))
print('Total de sapos: {}'.format(total_sapos))
porcentagem_coelhos = (total_coelhos * 100) / total
porcentagem_ratos = (total_ratos * 100) / total
porcentagem_sapos = (total_sapos * 100) / total
print('Percentual de coelhos: {:.2f} %'.format(porcentagem_coelhos))
print('Percentual de ratos: {:.2f} %'.format(porcentagem_ratos))
print('Percentual de sapos: {:.2f} %'.format(porcentagem_sapos))

File: Iniciante/1094.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int N, qntd;
    double pCoelho, pRatos, pSapos;
    double lista[3] = {};
    string tipo;
    cin >> N;
    for (int i=0; i<N; i++){
        cin >> qntd >> tipo;
        if (tipo == "C"){
            lista[0] += qntd;
        } else if (tipo == "R"){
            lista[1] += qntd;
        } else{
            lista[2] += qntd;
        }
    }
    pCoelho = (lista[0] * 100) / (lista[0] + lista[1] + lista[2]);
    pRatos = (lista[1] * 100 / (lista[0] + lista[1] + lista[2]));
    pSapos = (lista[2] * 100) / (lista[0] + lista[1] + lista[2]);
    cout << fixed << setprecision(0);
    cout << "Total: " << lista[0] + lista[1] + lista[2] << " cobaias" << endl;
    cout << "Total de coelhos: " << lista[0] << endl;
    cout << "Total de ratos: " << lista[1] << endl;
    cout << "Total de sapos: " << lista[2] << endl;
    cout << fixed << setprecision(2);
    cout << "Percentual de coelhos: " << pCoelho << " %" << endl;
    cout << "Percentual de ratos: " << pRatos << " %" << endl;
    cout << "Percentual de sapos: " << pSapos << " %" << endl;
    return 0;
}